packages/wonder-blocks-core/components/unique-id-provider.test.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import * as React from "react";
+import * as ReactDOMServer from "react-dom/server";
+
+import {mount} from "enzyme";
+
+import View from "./view.js";
+
+import SsrIDFactory from "../util/ssr-id-factory.js";
+import UniqueIDFactory from "../util/unique-id-factory.js";
+import UniqueIDProvider from "./unique-id-provider.js";
+
+describe("UniqueIDProvider", () => {
+    describe("mockOnFirstRender is default (false)", () => {
+        test("initial render is nothing on server", () => {
+            // Arrange
+            const children = jest.fn(() => null);
+            const nodes = <UniqueIDProvider>{children}</UniqueIDProvider>;
+
+            // Act
+            ReactDOMServer.renderToStaticMarkup(nodes);
+
+            // Assert
+            expect(children).not.toHaveBeenCalled();
+        });
+
+        test("initial render is nothing on client", () => {
+            // Arrange
+            const children = jest.fn(() => null);
+            const nodes = <UniqueIDProvider>{children}</UniqueIDProvider>;
+
+            // Act
+            mount(nodes);
+
+            // Assert
+            expect(children).toHaveBeenCalledTimes(1);
+            expect(children).not.toHaveBeenCalledWith(SsrIDFactory);
+        });
+
+        test("all renders get same unique id factory", () => {
+            // Arrange
+            const children = jest.fn(() => <View />);
+            const nodes = <UniqueIDProvider>{children}</UniqueIDProvider>;
+            const wrapper = mount(nodes);
+
+            // Act
+            wrapper.instance().forceUpdate();
```

Comment:
In practice what causes this update to occur on the client in normal use?

Reply:
A render. All we're doing here is forcing the render to occur again so that we can see we always get the same factory. In reality rather than forcing a render, something would have changed that caused react to re-render.